### PR TITLE
830 Parent menu items and <nolink>

### LINKED ISSUE #44: Parent menu items don't work correctly when not using <nolink>
Adding children to a menu item that has a link will confuse the menu system. The child menu indicator appears, on top of the link. Clicking the link will cause the mega menu to be displayed briefly and then the browser will navigate to the url of the parent item.

- [x] Render the child items indicator if, and only if, the <nolink> flag is used in place of a url.
- [x] For both the group menu and the regular menu, we need an additional if statement to check nolink in addition to the presence of children.
- [x] If the parent link is a hot link, ignore the children

See research and innovation link in attached screenshot or visit http://hansenm.dev.oregonstate.edu/ to view the bug on a site.

After further discussion with the product owner a few extra items have been added.

### PR DESCRIPTION
**Pull request by DeveloperPiatt**
See linked issues for more details

### CODE REVIEW

File: templates/navigation/menu--main.html.twig [hunk 1]
```diff
@@ -57,35 +57,41 @@ We call a macro which calls itself to render the full tree.
...
-      {% set data_index = data_index + 1 %}
+        <li{{ item.attributes.addClass(list_item_classes).setAttribute('data-index', data_index) }}>
+          {# Set accessibility featues to tab through menu and span to act as a button #}
+          {% if not item.url.isExternal() and item.url.getRouteName() is same as('<nolink>') and item.below %}
+            {{ link(item.title, item.url, item.attributes.addClass(link_classes).removeClass(list_item_classes).setAttribute('tabindex', 0).setAttribute('role', 'button')) }}
+          {% else %}
+            {{ link(item.title, item.url, {'class': link_classes}) }}
+          {% endif %}
+          {# adding the chevron #}
+{#          {% if menu_level == 0 and item.below and not item.url.isExternal() and item.url.getRouteName() is same as('<nolink>')%}#}
```
Comment (emerham) [inline templates/navigation/menu--main.html.twig]: If this is no longer going to be needed we should just remove it rather than comment it out.

**Review by emerham — CHANGES_REQUESTED** [review]
Just a question about one line if we should keep it there just for a little while for testing it's fine but we can always use git's history to go back and get code out to revert.

File: templates/navigation/menu--main.html.twig (hunk 1 above)
Comment (DeveloperPiatt) [reply templates/navigation/menu--main.html.twig]: agreed, i must have overlooked this. thanks

**Review by emerham — APPROVED** [review]
Ok LGTM

**Pull request merged**